### ISSUE TITLE
JupyterLab is crashing repeatedly when I try to run a large notebook on Chrome

### ISSUE THREAD

**Issue by kburchfiel** [jupyterlab/jupyterlab#16034]
## Description

I am attempting to run a large notebook (around 1,000 cells) within Chrome. In the process of running the notebook, its Chrome tab crashed around 5 or 6 times, leading to an 'Aw, Snap!' error page with a SIGILL error code.

Has anyone else experienced a similar issue, and if so, what steps would you recommend that I take to resolve it? I tried clearing the output of all of my cells, but this too produced an error message several times.

My laptop has 64 GB of RAM, so if the issue is related to the notebook's exceeding some predefined memory limit, perhaps I could resolve the issue by increasing that limit?

## Reproduce

2. Click on '...'
3. Scroll down to '...'
4. See error '...'

## Context

- Operating System and version: 
- Browser and version: 
- JupyterLab version: 

<details><summary>Troubleshoot Output</summary>
<pre>
Paste the output from running `jupyter troubleshoot` from the command line here.
You may want to sanitize the paths in the output.
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
Paste the output from your command line running `jupyter lab` here, use `--debug` if possible.
</pre>
</details>

<details><summary>Browser Output</summary>

<pre>
Paste the output from your browser Javascript console here, if applicable.

</pre>
</details>

**Comment by krassowski** [comment]
@kburchfiel which version of JupyterLab did you use?

Your issue might be a duplicate of https://github.com/jupyterlab/jupyterlab/issues/15465 which should have been fixed in the latest release, https://github.com/jupyterlab/jupyterlab/releases/tag/v4.1.5, by https://github.com/jupyterlab/jupyterlab/pull/15970.

**Comment by krassowski** [comment]
If you used the latest version/can reproduce it in the latest version, can you share the notebook which reproduces the issue and exact reproduction steps?

**Comment by kburchfiel** [comment]
According to `conda list` and the About JupyterLab window within JupyterLab, I'm running version 4.1.5. However, running `conda update jupyterlab` within my terminal did bring up a number of other libraries to update, so perhaps some of jupyterlab's dependencies weren't up-to-date? I went ahead and updated those libraries, so I'll see if that resolves the issue.

I also tried running `conda update jupyterlab` within my base environment, but conda ended up hanging indefinitely. I'm not sure why the update command would work for a virtual environment but not a base one, but perhaps this issue is related to my original question. 

I'm not able to share this file in particular, but if I see similar behavior within a file that I *can* share publicly, I'll make sure to do so.

I did try running JupyterLab in Firefox and didn't see the same behavior, so perhaps the issue is related to Chrome?

**Comment by krassowski** [comment]
> so perhaps some of jupyterlab's dependencies weren't up-to-date

No, the fix was in the `jupyterlab` package itself, so you should be good.

> I'm not able to share this file in particular, but if I see similar behavior within a file that I can share publicly, I'll make sure to do so.

That would be appreciated!

> I'd still like to figure out why the program is having issues in Chrome, but if anyone else is experiencing issues with large notebooks, consider running your notebook in Firefox instead

This is indeed something that I noted that the crashes only occurred in Chromium based browsers (I saw it in Chrome and Edge). However, I did less testing in Firefox.

Did you use `full` windowing mode by any chance? (Settings → Notebook → Windowing mode)

**Comment by kburchfiel** [comment]
Thanks for the prompt response! I'm testing out full windowing mode within JupyterLab on Chrome, and so far the notebook is working great. I'm not seeing any crashes or SIGILL errors appear, and the performance is fine also. 

I'll check to see which windowing mode had been set on Firefox, but I believe it was 'defer', since I hadn't made any changes to the default windowing mode there.

So it does seem that the 'defer' windowing mode causes issues in at least some large notebooks in Chrome (but perhaps not Firefox), but these can be resolved by changing the mode to 'full.' 

I can of course also try to reduce the amount of output the notebook is presenting; it's not like I'm reviewing all 1000 cells each time I run it. But it looks like this step may not be necessary as long as I keep the full windowing mode in place.

**Comment by JasonWeill** [comment]
@kburchfiel Thank you for opening this issue! If you could share with us a sample notebook that causes this problem to recur, that would help a developer work on a fix. Even describing the notebook (numbers of cells, outputs, visual outputs, etc) would help give us perspective.

This also represents an opportunity to improve our documentation to suggest size limits for notebooks out of performance concerns. This is likely to vary widely based on notebook type and computer type.

#15258 concerns enabling "full" windowing notebook mode by default (thanks @RRosio )

**Comment by sina-mansour** [comment]
I'm also facing this similar issue, In my case, I have several notebooks open (more than 10), and some of the notebooks have more than 100 cells. Most importantly, some notebooks have a considerable size (200-300MB) as they contain several figures.

Today, it started crashing every now and then with the sigill error code in chrome, I'm not sure what's causing this as the same setup used to work for a long time (the only thing that I can think of is if I have possibly passed a certain memory limit) without any warnings, I should nonetheless highlight that both the server and the client have plenty of unused RAM, so memory shouldn't be the cause of crash, however I thought It would be helpful if I share that. there are no errors on the server side (the crash is only on the client side, i.e. the chrome tab). Notably, before the crash, the chrome tab with jupyter lab running consumes upwards of 5GB RAM seen from chrome's task manager.

Finally, I speculate it might be during notebook save operations that the crash occurs (just a guess, but not really sure).

**Comment by krassowski** [comment]
@sina-mansour which exact version are you using? Any extensions which could be contributing to the memory usage?

**Comment by sina-mansour** [comment]
I'm running Version 3.6.7, and have the following extensions installed. But I don't have any reason to believe any extensions might be causing this, as none were installed recently.

```
🔥 jupyter --version
Selected Jupyter core packages...
IPython          : 8.20.0
ipykernel        : 6.28.0
ipywidgets       : 8.0.7
jupyter_client   : 7.4.9
jupyter_core     : 5.5.0
jupyter_server   : 2.10.0
jupyterlab       : 3.6.7
nbclient         : 0.8.0
nbconvert        : 7.10.0
nbformat         : 5.9.2
notebook         : 6.5.4
qtconsole        : not installed
traitlets        : 5.7.1
```

<img width="270" height="687" alt="Image" src="https://github.com/user-attachments/assets/9e3eff53-27dd-48de-92ba-11545581cf28" />

Is this something that might be due to an outdated version?

---

P.S. I will try updating to a more recent version to see if that fixes the issue.

**Comment by krassowski** [comment]
Newer versions do include some fixeswhich could help:
- https://github.com/jupyterlab/jupyterlab/pull/17628
- https://github.com/jupyterlab/jupyterlab/pull/17707

I would suggest trying the 4.5.0b1 already since we will have an RC next week and realease this as final before or during JupyterCon.

**Comment by sina-mansour** [comment]
Ok, so I updated to a fresh conda install of the following:

🔥 jupyter --version                                                                                                                                                                          
Selected Jupyter core packages...                                                                                                                                                             
IPython          : 8.37.0                                                                                                                                                                     
ipykernel        : 6.29.5                                                                                                                                                                     
ipywidgets       : 8.1.7                                                                                                                                                                      
jupyter_client   : 8.6.3                                                                                                                                                                      
jupyter_core     : 5.8.1                                                                                                                                                                      
jupyter_server   : 2.17.0                                                                                                                                                                     
jupyterlab       : 4.4.10                                                                                                                                                                     
nbclient         : 0.10.2                                                                                                                                                                     
nbconvert        : 7.16.5                                                                                                                                                                     
nbformat         : 5.10.4                                                                                                                                                                     
notebook         : not installed                                                                                                                                                              
qtconsole        : not installed                                                                                                                                                              
traitlets        : 5.14.3

and this seems to have reduced the memory usage (to ~2GB) and have not had a crash in the past 30 minutes, which is a good sign.

Thanks for the suggestions. 👍
